contracts/contracts/strategies/balancer/BaseBalancerStrategy.sol

```diff
@@ -477,13 +499,43 @@ abstract contract BaseBalancerStrategy is InitializableAbstractStrategy {
 
     function _approveBase() internal virtual {
         IERC20 pToken = IERC20(platformAddress);
+
         // Balancer vault for BPT token (required for removing liquidity)
         pToken.safeApprove(address(balancerVault), type(uint256).max);
     }
 
+    /**
+     * @notice Returns the rate supplied by the Balancer configured rate
+     * provider. Rate is used to normalize the token to common underlying
+     * pool denominator. (ETH for ETH Liquid staking derivatives)
+     *
+     * @param _asset Address of the Balancer pool asset
+     * @return rate of the corresponding asset
+     */
     function _getRateProviderRate(address _asset)
         internal
         view
         virtual
-        returns (uint256);
+        returns (uint256)
+    {
+        IBalancerPool pool = IBalancerPool(platformAddress);
+        IRateProvider[] memory providers = pool.getRateProviders();
```

Comment:
Any reason for not doing `IRateProvider[] memory providers = IBalancerPool(pool).getRateProviders();` since `pool` is used only once?

Reply:
Thanks both very good points. Have reduced the code to 1 line. The caching is done in this commit: https://github.com/OriginProtocol/origin-dollar/pull/1839/commits/8c8732004095f7c29ea7b569f6a45792f3dce62b

(the rate providers of a pool are not expected to change)